src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -1554,3 +1556,101 @@ bool IdealLoopTree::loop_predication( PhaseIdealLoop *phase) {
 bool IdealLoopTree::can_apply_loop_predication() {
   return _head->is_Loop() && !_irreducible && !tail()->is_top();
 }
+
+bool PhaseIdealLoop::is_uncommon_trap_if_pattern(IfProjNode* proj) {
+  if (proj->is_uncommon_trap_if_pattern()) {
```

Comment:
Is having two methods with an identical name but subtly different semantics not quite confusing, and maybe going to lead to some subtle bugs later on?

Reply:
Thanks for the update :heavy_check_mark: